Add ConfigProvider

geerteltink opened the pull request:
Adds ConfigProvider with basic configuration.

geerteltink committed:
Added ConfigProvider with basic configuration

geerteltink committed:
Bring composer.lock insync with composer.json

michalbundyra commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-fastroute for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
Only 2018 for new files, please.

michalbundyra commented on composer.json:
```diff
@@ -55,6 +55,9 @@
             "dev-release-3.0.0": "3.0.x-dev"
+        },
+        "zf": {
+            "config-provider": "Zend\\Expressive\\Router\\ConfigProvider"
```
Maybe we should consider change namespace in the package to have it consistent with repository name.
Now it is so confusing, and we are using the same namespace in coupe repositories...

weierophinney replied:
Doing so would be quite problematic from a migration stand-point, however.

That said: @xtreamwayz We should likely put the `ConfigProvider` classes for the routers and template providers into a subnamespace in order to prevent collisions: `Zend\Expressive\Router\FastRoute\ConfigProvider`.

webimpress commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-fastroute/blob/master/LICENSE.md New BSD License
+ */
+
```
No new line before strict_type declaration

michalbundyra commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+    {
+        return [
+            'factories' => [
+                RouterInterface::class => FastRouteRouterFactory::class,
```
I would suggest here to have:

```php
'aliases' => [
    RouterInterface::class => FastRouteRouter::class,
],
'factories' => [
    FastRouteRouter::class => FastRouteRouterFactory::class,
],
```

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-fastroute for the canonical source repository
+ * @copyright Copyright (c) 2017-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
Only 2018 please (new file)

webimpress commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+ * @copyright Copyright (c) 2017-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-fastroute/blob/master/LICENSE.md New BSD License
+ */
+
```
No new line before strict types declaration

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+    public function testInvocationReturnsArray() : array
+    {
+        $config = ($this->provider)();
+        self::assertInternalType('array', $config);
```
$this->.... ?

weierophinney replied:
There's a forum topic discussing this currently. I think for now, we should use `$this->`, as it's consistent with every other component. If we decide to go with `self::` later as a result of that forum topic, we can apply that via a phpcs fixer.

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+        self::assertInternalType('array', $config['dependencies']);
+        self::assertArrayHasKey('factories', $config['dependencies']);
+        self::assertInternalType('array', $config['dependencies']['factories']);
+        self::assertArrayHasKey(RouterInterface::class, $config['dependencies']['factories']);
```
$this->... 5x ?

weierophinney committed:
Update to PSR-15

Updates component to pin to zend-expressive-router 3.0.0alpha1 or above,
which means using PSR-15 for middleware. The only change necessary in
this component was to use the PSR-15 `MiddlewareInterface` in unit
tests, versus the http-interop variant.

weierophinney committed:
Adds CHANGELOG entry for #46

weierophinney committed:
Merge branch 'feature/psr-15' into release-3.0.0

Close #46

weierophinney committed:
Merge pull request #45 from xtreamwayz/feature/config-provider

Add ConfigProvider

Conflicts:
	composer.lock

weierophinney committed:
Moves ConfigProvider to subnamespace

In order to prevent conflicts with other router implementations, moves
the `ConfigProvider` to a new subnamespace,
`Zend\Expressive\Router\FastRouteRouter\ConfigProvider`.

weierophinney committed:
Updates copyright ranges to reflect file additions

weierophinney committed:
Define router interface as an alias

Per @webimpress, defines `RouterInterface` as an alias, pointing to the
FastRouteRouter service, which in term points to its factory.

weierophinney committed:
Use consistent assertion style

We use `$this->` for assertions basically everywhere else in our
components. We can change to `self::` at a later date.

weierophinney commented:
Thanks, @xtreamwayz!
